factoryWithTypeCheckers.js

```diff
@@ -302,8 +302,13 @@ module.exports = function(isValidElement, throwOnDirectAccess) {
         }
       }
 
-      var valuesString = JSON.stringify(expectedValues);
-      return new PropTypeError('Invalid ' + location + ' `' + propFullName + '` of value `' + propValue + '` ' + ('supplied to `' + componentName + '`, expected one of ' + valuesString + '.'));
+      var valuesString = JSON.stringify(expectedValues, function replacer(key, value) {
+        if (getPropType(value) === 'symbol') {
```

Comment:
Why not unconditionally use String here?

Reply:
In the template literal they’ll all be converted to a string anyways; it seems simpler to avoid the symbol check.

Reply:
For clarity, am I correct that you're suggesting the following?

```js
var valuesString = JSON.stringify(expectedValues, function replacer(key, value) {
  return String(value);
});
```

I just want to point out that this changes the warning output slightly:

```
// Before:
... expected one of ["red","blue"].

// After:
... expected one of "red,blue".
```

Thoughts?

Reply:
oh right, because of objects and arrays. i guess as-is is better, then.